signer/web/src/reducers/transactions.js

```diff
@@ -15,33 +15,80 @@ export default handleActions({
     };
   },
 
-  'add confirmedTransaction' (state, action) {
-    const pending = state.pending.filter(tx => tx.id !== action.payload.id);
+  'start confirmTransaction' (state, action) {
+    const pending = state.pending.map(p => {
+      if (p.id === action.payload.id) {
+        p.isSending = true;
+      }
+      return p;
+    });
+    return {
+      ...state,
+      pending
+    };
+  },
 
+  'success confirmTransaction' (state, action) {
+    const { id, txHash } = action.payload;
+    const confirmed = Object.assign({},
+      state.pending.find(p => p.id === id),
+      { txHash }, { status: 'confirmed' }
```

Comment:
1. `state.pending.find(...)` would mutate the state
2. `{ status: 'confirmed', txHash }` :+1: